mysql-test/suite/innodb/r/innodb-index.result

```diff
@@ -1947,3 +1947,63 @@ Warnings:
 Warning	1082	InnoDB: Table test/t contains 0 indexes inside InnoDB, which is different from the number of indexes 1 defined in the MariaDB 
 Warning	1082	InnoDB: Table test/t contains 0 indexes inside InnoDB, which is different from the number of indexes 1 defined in the MariaDB 
 DROP TABLE t;
+#
+#  MDEV-35115 Inconsistent Replace behaviour when multiple
+#               unique index exist
+#
+CREATE TABLE t1 (c1 NUMERIC UNSIGNED NOT NULL,
+c2 INT3 UNIQUE,
+c3 BIT(2) PRIMARY KEY)ENGINE=InnoDB;
+CREATE UNIQUE INDEX i1 ON t1(c1);
+INSERT INTO t1 (c1,c2,c3) VALUES (0,0,b'01');
+INSERT INTO t1 (c1,c2,c3) VALUES (1,1,b'10');
+FLUSH STATUS;
+SELECT * FROM INFORMATION_SCHEMA.SESSION_STATUS WHERE VARIABLE_NAME LIKE 'HANDLER_%' AND VARIABLE_VALUE > 0;
+VARIABLE_NAME	VARIABLE_VALUE
+HANDLER_TMP_WRITE	24
+REPLACE INTO t1 (c1,c2,c3) VALUES (0,1,b'11');
+SELECT * FROM INFORMATION_SCHEMA.SESSION_STATUS WHERE VARIABLE_NAME LIKE 'HANDLER_%' AND VARIABLE_VALUE > 0;
+VARIABLE_NAME	VARIABLE_VALUE
+HANDLER_COMMIT	1
+HANDLER_DELETE	2
+HANDLER_READ_KEY	2
+HANDLER_READ_RND_NEXT	28
+HANDLER_TMP_WRITE	51
+HANDLER_WRITE	3
+ALTER TABLE t1 ALGORITHM=COPY;
+FLUSH STATUS;
+SELECT * FROM INFORMATION_SCHEMA.SESSION_STATUS WHERE VARIABLE_NAME LIKE 'HANDLER_%' AND VARIABLE_VALUE > 0;
+VARIABLE_NAME	VARIABLE_VALUE
+HANDLER_TMP_WRITE	24
```

Comment:
It looks like `HANDLER_TMP_WRITE` is totally irrelevant for this test. I’d suggest to filter it out in the `WHERE` condition. Perhaps we should enumerate the conditions that we are really interested in, namely, `WHERE VARIABLE_NAME IN ('HANDLER_DELETE','HANDLER_WRITE','HANDLER_READ_KEY')`. In that way, even if later branches change something around the irrelevant counters, the results of this tests will not change.